cpp/ql/test/query-tests/Security/CWE/CWE-134/semmle/ifs/ifs.expected

```diff
@@ -1,157 +1,157 @@
 edges
+| ifs.c:61:8:61:11 | argv | ifs.c:61:8:61:14 | access to array |
+| ifs.c:61:8:61:11 | argv | ifs.c:62:2:62:7 | Phi |
 | ifs.c:61:8:61:11 | argv | ifs.c:62:9:62:10 | (const char *)... |
-| ifs.c:61:8:61:11 | argv | ifs.c:62:9:62:10 | (const char *)... |
-| ifs.c:61:8:61:11 | argv | ifs.c:62:9:62:10 | c7 |
-| ifs.c:61:8:61:11 | argv | ifs.c:62:9:62:10 | c7 |
 | ifs.c:61:8:61:11 | argv | ifs.c:62:9:62:10 | c7 |
 | ifs.c:61:8:61:11 | argv | ifs.c:62:9:62:10 | c7 |
+| ifs.c:68:8:68:11 | argv | ifs.c:68:8:68:14 | access to array |
+| ifs.c:68:8:68:11 | argv | ifs.c:69:2:69:7 | Phi |
 | ifs.c:68:8:68:11 | argv | ifs.c:69:9:69:10 | (const char *)... |
-| ifs.c:68:8:68:11 | argv | ifs.c:69:9:69:10 | (const char *)... |
-| ifs.c:68:8:68:11 | argv | ifs.c:69:9:69:10 | c8 |
 | ifs.c:68:8:68:11 | argv | ifs.c:69:9:69:10 | c8 |
 | ifs.c:68:8:68:11 | argv | ifs.c:69:9:69:10 | c8 |
-| ifs.c:68:8:68:11 | argv | ifs.c:69:9:69:10 | c8 |
-| ifs.c:74:8:74:11 | argv | ifs.c:75:9:75:10 | (const char *)... |
+| ifs.c:74:8:74:11 | argv | ifs.c:74:3:74:14 | Store |
+| ifs.c:74:8:74:11 | argv | ifs.c:74:8:74:14 | access to array |
 | ifs.c:74:8:74:11 | argv | ifs.c:75:9:75:10 | (const char *)... |
 | ifs.c:74:8:74:11 | argv | ifs.c:75:9:75:10 | i1 |
 | ifs.c:74:8:74:11 | argv | ifs.c:75:9:75:10 | i1 |
-| ifs.c:74:8:74:11 | argv | ifs.c:75:9:75:10 | i1 |
-| ifs.c:74:8:74:11 | argv | ifs.c:75:9:75:10 | i1 |
+| ifs.c:80:8:80:11 | argv | ifs.c:80:3:80:14 | Store |
+| ifs.c:80:8:80:11 | argv | ifs.c:80:8:80:14 | access to array |
 | ifs.c:80:8:80:11 | argv | ifs.c:81:9:81:10 | (const char *)... |
-| ifs.c:80:8:80:11 | argv | ifs.c:81:9:81:10 | (const char *)... |
-| ifs.c:80:8:80:11 | argv | ifs.c:81:9:81:10 | i2 |
```

Comment:
Why are these display strings changing? I'd like to avoid IR opcode names in these user-visible strings.

Reply:
This problem seems to have gone away now.

Reply:
I think those were part of the unrestricted flow graph coming from an indirect import of `DefaultTaintTracking`. The underlying problem is still there - if an `InstructionNode` doesn't compute the result of an expression, and a description for it isn't provided by another subclass like `ThisParameterNode`, then the opcode name is used. We could just use the `getAST` always, but I think that would be misleading in some cases.

Reply:
I agree. Let's keep the "ugly" names in place and then address them when we see them in practice.